backtester/report/report_test.go

```diff
@@ -290,7 +292,21 @@ func TestGenerateReport(t *testing.T) {
 			},
 		},
 	}
-	err := d.GenerateReport()
+	timeNow := time.Now()
+	err := d.GenerateReport(timeNow)
+	if err != nil {
+		t.Error(err)
+	}
+	var nickName string
+	if d.Config.Nickname != "" {
+		nickName = d.Config.Nickname + "-"
+	}
+	fileName := fmt.Sprintf(
+		"%v%v-%v.html",
+		nickName,
+		d.Statistics.StrategyName,
+		timeNow.Format("2006-01-02-15-04-05"))
```

Comment:
This can be done without needing to change the function signature to have a `time.Time`. `d` has a property `OutputPath` which we can use to set to a temporary directory. Then we just delete everything inside that directory
```
tempDir, err := ioutil.TempDir("", "")
	if err != nil {
		t.Fatalf("Problem creating temp dir at %s: %s\n", tempDir, err)
	}
	defer os.RemoveAll(tempDir)
d.OutputPath = tempDir
```